PWGHF/D2H/Tasks/taskDs.cxx

```diff
@@ -227,54 +228,58 @@ struct HfTaskDs {
   {
     // MC rec.
     for (auto& candidate : candidates) {
-      if (yCandMax >= 0. && std::abs(yDs(candidate)) > yCandMax) {
-        continue;
-      }
-      if (std::abs(candidate.flagMcMatchRec()) == 1 << DecayType::DsToKKPi) {
-        auto prong0McPart = candidate.prong0_as<aod::BigTracksMC>().mcParticle_as<candDsMcGen>();
-        auto indexMother = RecoDecay::getMother(particlesMC, prong0McPart, pdg::Code::kDS, true);
-        auto particleMother = particlesMC.iteratorAt(indexMother);
-        registry.fill(HIST("hPtGenSig"), particleMother.pt()); // gen. level pT
-
-        // KKPi
-        if (std::abs(prong0McPart.pdgCode()) == kKPlus) {
-          fillHistoMCRec(candidate, candidate.isSelDsToKKPi());
+      if (candidate.flagMcDecayChanRec() == Dsdecaychannel) {
```

Comment:
Add this after the check for the candidate to be matched to MC, line 235 in your current code. 
Also, use the same form as the rapidity check here below that is better for indentation.